File: HotelManagementFrontend/src/app/services/booking.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {BookingRequest} from "../models/BookingRequest";
import {Observable} from "rxjs";
import {Booking} from "../models/Booking";
import {BookingWithDetails} from "../models/BookingWithDetails";

@Injectable({
  providedIn: 'root'
})
export class BookingService {
  readonly baseURL = "http://localhost:8888/bookings";

  constructor(private httpClient: HttpClient) {
  }

  public add(bookingRequest: BookingRequest): Observable<Booking> {
    const headers = {'Content-Type': 'application/json'};
    return this.httpClient.post<Booking>(this.baseURL, bookingRequest, {headers});
  }

  public cancel(bookingId: number): Observable<any> {
    const headers = {'Content-Type': 'application/json'};
    return this.httpClient.put<any>(this.baseURL + '/' + bookingId, {}, {headers});
  }

  public updateRoom(bookingId: number, roomId: number): Observable<Booking> {
    const headers = {'Content-Type': 'application/json'};
    return this.httpClient.put<Booking>(this.baseURL + '/' + bookingId + '/' + roomId, {}, {headers});
  }

  public getAllByUserId(userId: number): Observable<BookingWithDetails[]> {
    const headers = {'Content-Type': 'application/json'};
    return this.httpClient.get<BookingWithDetails[]>(this.baseURL + '/' + userId, {headers});
  }
}

File: HotelManagementBackend/src/main/java/com/internship/hotelmanagementbackend/service/BookingService.java
package com.internship.hotelmanagementbackend.service;

import com.internship.hotelmanagementbackend.dto.BookingDto;
import com.internship.hotelmanagementbackend.dto.BookingRequestDto;
import com.internship.hotelmanagementbackend.dto.BookingWithDetailsDto;

import java.util.List;

public interface BookingService {
    BookingDto add(BookingRequestDto bookingRequestDto);

    void cancel(Long id);

    BookingDto updateRoom(Long bookingId, Long roomId);

    List<BookingWithDetailsDto> getAllByUserId(Long userId);
}
